pyznap/send.py

```diff
@@ -316,7 +316,7 @@ def send_config(config):
                 # send not excluded filesystems
                 for retry in range(1,retries+2):
                     rc = send_filesystem(source_fs, dest_name, ssh_dest=ssh_dest, raw=raw, resume=resume)
-                    if rc == 2 and retry <= retries:
+                    if rc != 0 and retry <= retries:
```

Comment:
We will only get rc==2 in the first fail of network. In the first retry, we will get other others errors(and the loop return 1). The code will not reach to the `send` and `receive` command.
so just return 1 when we got a error, continue the retry loop when we get the error
```
# test before this change
pyznap$ pyznap send
Sep 19 21:45:15 INFO: Starting pyznap...
Sep 19 21:45:15 INFO: Sending snapshots...
Sep 19 21:45:15 WARNING: lzop does not exist, continuing without compression...
.....
Sep 19 21:45:16 INFO: Found resume token. Resuming last transfer of t/backups/remote/t/large (~154.1M)...
Sep 19 21:45:39 ERROR: Error while sending to t/backups/remote/t/large: cannot receive resume stream: checksum mismatch or incomplete stream. - Partially received snapshot is saved. - A resuming stream can be generated on the sending system by running: -     zfs send -t 1-b9364b1bd-b8-789c636064000310a500c4ec50360710e72765a52697303030419460caa7a515a7968064c45860f26c48f2499525a9c5208989322cd8f497e4a79766a63030cc9428547912d7b94f03499e132c9f97989b0aa4f573128bd2531d8c0c0c7ca14e610000de681599...
Sep 19 21:45:39 INFO: Retrying send in 10s (retry 1 of 10)...
Sep 19 21:45:49 ERROR: Error while opening source root@remote:t/large: 'ssh: Could not resolve hostname 8027.node.office: Name or service not known'...
# no more tries because the code return 1 before it reach `send` and `receive` and return 2
Sep 19 21:45:49 INFO: Finished successfully...
```

Reply:
I specifically set it to `rc == 2` because I don't want to trigger retries for all errors. E.g. when there are no snapshots on the source the method `send_filesystem` will return 1, but we don't want to retry on that. I'll revert the check to `rc == 2` but will check for ssh errors especially as you suggested in the beginning.